File: v2/src/main/java/com/api/v2/doctors/services/DoctorRetrievalService.java
package com.api.v2.doctors.services;

import com.api.v2.doctors.dtos.DoctorResponseDto;
import de.kamillionlabs.hateoflux.model.hal.HalResourceWrapper;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface DoctorRetrievalService {
    Mono<HalResourceWrapper<DoctorResponseDto, Void>> findByMedicalLicenseNumber(String medicalLicenseNumber);
    Flux<DoctorResponseDto> findAll();
}

File: v2/src/main/kotlin/com/api/v2/doctors/services/DoctorRetrievalService.kt
package com.api.v2.doctors.services

import com.api.v2.doctors.responses.DoctorResponseDTO
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.http.ResponseEntity

interface DoctorRetrievalService {
    fun findByLicenseNumber(licenseNumber: String): ResponseEntity<DoctorResponseDTO>
    fun findAll(pageable: Pageable): ResponseEntity<Page<DoctorResponseDTO>>
}